philipmeneghini/InstaConnect
# Extend Session Window

**Summary**
Using the `createContext` and `useContext` hook I was able to create the component `UserProvider` which wraps around the application. This component is used to monitor changes in the jwt stored in local storage. It then decodes and retrieves the `UserModel` associated with that jwt. This way I can access the user who is logged in all across my application. Additionally I deployed a `useEffect` which tracks when the token is expected to expire. When the jewt has only five minutes or less in its lifetime a popup is displayed to direct the user to either logout or refresh their session.

![ExtendModal](https://github.com/philipmeneghini/InstaConnect/assets/63440066/5a99ce05-d586-4ec7-95ae-12f55f274b6c)

**Testing**

- jwt is set to expire in an hour check that in an hour of being logged in that the popup appears to refresh the session
- Try logging out and check that you are correctly redirected
- Check that the correct user is being retrieved when logged in. `useContext` hook should always retrieve that for all components
